### ISSUE TITLE
Rule proposal: variable-right-hand-side

### ISSUE THREAD

**Issue by dj-wong** [eslint/eslint#8406]
**Please describe what the rule should do:**
Ensure at least one token on the right-hand-side is on the same line as the variable declaration.

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
```js
var a = 
"test string";

var b = 
2;
```

```js
// object literals
let a = 
{ foo:  1};

let c = 
{
  foo: 1, bar: 2
};

let d = 
{
  foo: 1,
  bar: 2
};
```

```js
// array literals
var a = 
[ 1, 2, 3 ];
```

```js
// function expressions
var someFunction = 
function() 
{
  // some code
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Currently, there are only rules checking for newlines within the braces/brackets (object-curly-newline, object-property-newline, and array-bracket-newline #6073, and brace-style to some extent).  To have a wider coverage in code-styling in this area, a new styling rule enforcing the above would be beneficial.

**Comment by platinumazure** [comment]
Hi @dj-wong, thanks for creating an issue.

I'm basically :+1: to the idea, but we might need to bikeshed a bit about the rule name and API.

The basic idea, as far as I understand it, is we should be able to require or forbid a newline after the equals sign when assigning to a variable (either in an AssignmentExpression or a VariableDeclarator). We should try to think about what sort of styles might be preferred.

Off the top of my head, here is what I could envision us supporting:
* No newline after equals sign, ever (i.e., require at least one token on the same line, same as what @dj-wong wants)
* Allow newline only if (or require iff) the RHS of the equals sign needs multiple lines:

    ```js
    var a =
    [
        "foo"
    ];
    ```
* Always require a newline (I guess? Not sure if there is value here):

    ```js
    var a =
        "foo";
    ```

Maybe I'm overthinking it, but we don't like writing rules that are designed for only one style and I want to make sure the rule is flexible for core.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
